tldr.py

```diff
@@ -504,6 +508,40 @@ def main() -> None:
                 with open(command, encoding='utf-8') as open_file:
                     output(open_file.read().encode('utf-8').splitlines(),
                            plain=options.markdown)
+    elif options.search:
+        command = '-'.join(options.command)
+        page = None
+        maxprob = 0
+        searchquery = options.search.split(' ')
+
+        for i in get_commands():
+            if i.startswith(command):
+                for p in get_platform_list():
+                    result = load_page_from_cache(i, p, options.language)
+                    if result is not None and have_recent_cache(i, p, options.language):
+                        break
+                try:
+                    result = result.decode("utf-8")
+                    result = result.split()
+                except AttributeError:
+                    print("Please update cache")
```

Comment:
This exception message does not make sense to me. It's trying to protect from a broken cache? Does this happen in meaningful fashion?

Reply:
If the user didn't update the cache, the program would take a long time trying to get the pages from the Internet one by one, and this was a way to prevent that happening and explain to the user to update the cache

Reply:
Oh, I get it now. Can you replace the `try / except` here with a `if result is None:` check as that will be a lot clearer (to me) that the intent is handling the case where we did not find any matching result page.

Reply:
Yea, I've added that in the new commit